code/game/turfs/simulated/chasm.dm

```diff
@@ -139,6 +139,9 @@
 		qdel(S.mmi)
 
 	qdel(AM)
-
+	
+	if(AM && !QDELETED(AM))	//It's indestructible
+		AM.throw_at(get_edge_target_turf(src,pick(alldirs)),rand(1, 10),rand(1, 10))
```

Comment:
also you gotta update_atom_color and reset its alpha and pixel_y or your object will be invisible.